website/static/js/filepage/index.js

```diff
@@ -17,6 +18,65 @@ var Panel = utils.Panel;
 
 var EDITORS = {'text': FileEditor};
 
+var MakeClient = function(element) {
+    makeClient(element);
+};
+
+var CopyButton = {
+    view: function(ctrl, params) {
+        return m('span.input-group-btn', m('button#copyBtn.btn.btn-default.btn-md[type="button"][style="height:' + params.copyButtonHeight + '"][data-clipboard-text="' + params.link + '"]',
+            {config: MakeClient},
+            m('.fa.fa-copy')));
+    }
+};
+
+var SharePopover =  {
+    view: function(ctrl, params) {
+        var copyButtonHeight = '34px';
+        var popoverWidth = '450px';
+        var link = params.link.substring(0, params.link.indexOf('download') + 8) + '%26mode=render';
```

Comment:
Discussed in person; this cleaning logic should probably go in `FileViewPage.view`. Also, you should handle the case where `link` may be undefined. This might happen if the MFR service is down. With the current code I get a JS error:

```
Uncaught TypeError: Cannot read property 'substring' of undefined
```

You can reproduce this by turning your MFR server off and going to a file detail page.